src/modules/AutoinstSoftware.rb

```diff
@@ -201,16 +203,27 @@ def Import(settings)
 
     def AddYdepsFromProfile( entries )
       Builtins.y2milestone("AddYdepsFromProfile entries %1", entries)
-      ign = [ "software", "partitioning", "general", "report", "scripts", "suse_register", 
-              "files", "bootloader", "add-on", "dasd", "zfcp", "timezone" ]
-      map = { "networking" => "yast2-network" }
-      entries.reject! { |e| ign.include?(e) }
-      Builtins.y2milestone("AddYdepsFromProfile entries %1", entries)
       pkglist = []
       entries.each do |e|
-        name = map.has_key?(e) ? map[e] : "yast2-"+e
-        Builtins.y2milestone("AddYdepsFromProfile name %1 from %2", name, e)
-        pkglist.push(name) if !pkglist.include?(name)
+        yast_modules = Y2ModuleConfig.ModuleMap.select {|module_name, entry|
+          module_name == e ||
+          entry["X-SuSE-YaST-AutoInstResource"] == e ||
+          (entry["X-SuSE-YaST-AutoInstMerge"] && entry["X-SuSE-YaST-AutoInstMerge"].split(",").include?(e))
+        }
+        if yast_modules.empty?
+          # taking default because no entry has been defined in the *.desktop file
+          provide = "application(YaST2/#{e}.desktop)"
+        else
+          provide = "application(YaST2/#{yast_modules.keys.first}.desktop)"
+        end
```

Comment:
you use yast_modules only first element and from code it looks like, there is only one matching. also structure looks similar, so maybe code can look like this:
```
yast_module, _entry = Y2ModuleConfig.ModuleMap.find do |module_name, entry|
  module_name == e ||
 entry["X-SuSE-YaST-AutoInstResource"] == e ||
 (entry["X-SuSE-YaST-AutoInstMerge"] && entry["X-SuSE-YaST-AutoInstMerge"].split(",").include?(e))
end
yast_module ||= e
provide = "application(YaST2/#{yast_module}.desktop)"
```